client/handlers.go

```diff
@@ -133,9 +133,10 @@ func (v2 *signer) Sign() error {
 	v2.Request.Header["x-amz-date"] = []string{v2.Time.In(time.UTC).Format(time.RFC1123)}
 
 	// Alibaba Cloud OSS date's formate must be http.TimeFormat
-	// URL Host's format is host or host:port
-	if config.Provider(strings.Split(host, ":")[0]) == "alicloud" {
+	// Alibaba Cloud OSS uses virtual hosted and the URL Host's format is <bucket-name>.host or <bucket-name>host:port
+	if config.Provider(strings.Join(strings.Split(strings.Split(host, ":")[0], ".")[1:], ".")) == "alicloud" {
 		v2.Request.Header["x-amz-date"] = []string{v2.Time.In(time.UTC).Format(http.TimeFormat)}
+		canonicalPath = fmt.Sprintf("/%s%s", strings.Split(host, ".")[0], canonicalPath)
 	}
```

Comment:
@xiaozhu36 looks like this is worth some tests ?

Reply:
HI @pivotal-jamil-shamy Thanks for your feedback. I have added a test case #20 for alibaba cloud. Comparing with previous usage, this PR will not have any difference. So, I think #20 can also apply for this PR.